### PR TITLE
Add -out parameter to /CAT

### PR DESCRIPTION
**Pull request by vague666**

### CODE REVIEW

**Comment by ailin-nemui** [comment]
btw, /exec -o cat

**Comment by ailin-nemui** [comment]
this is confusing because of the name -out, maybe use "-window" or such?

**Comment by dequis** [comment]
Is this intended to write to the channel publicly as `/exec -out`, or to write to the current window locally as `/echo`? Because the help string talks about it as the former, and the code behaves as the latter.

Also I don't want to bikeshed but `-window` is not a great name, as it's usually a parameter that takes a value (like `/echo -window <name>`).

Ironically -current, which #1024 removes, would make it consistent with /echo (unchangeable default behavior aside)

**Comment by vague666** [comment]
-window is used for various purposes, /exec -window sends the output to active window for instance, while /echo -window takes an argument. /exec -out sends the output as a msg to the active window which we might not want to mix into this

**Comment by dequis** [comment]
Oh, okay, then you probably want to fix the help text which has the same wording as /exec -out

File: docs/help/in/cat.in [hunk 1]
```diff
@@ -5,19 +5,22 @@
 
 %9Parameters:%9
 
+    -window:            Prints the output int the active window.
```
Comment (dequis) [inline docs/help/in/cat.in]: void

**Comment by dequis** [comment]
Sounds fine to me

File: src/fe-common/core/fe-core-commands.c [hunk 2]
```diff
@@ -152,15 +155,18 @@ static void cmd_cat(const char *data)
...
 
+	target = g_hash_table_lookup(optlist, "window") != NULL ? item->name : NULL;
+
 	g_io_channel_set_encoding(handle, NULL, NULL);
 	g_io_channel_seek_position(handle, fpos, G_SEEK_SET, NULL);
 	buf = g_string_sized_new(512);
 	while (g_io_channel_read_line_string(handle, buf, &tpos, NULL) == G_IO_STATUS_NORMAL) {
 		buf->str[tpos] = '\0';
-		printtext(NULL, NULL, MSGLEVEL_CLIENTCRAP |
+		printtext(NULL, target, MSGLEVEL_CLIENTCRAP |
```
Comment (ailin-nemui) [inline src/fe-common/core/fe-core-commands.c]: if -window is given, this should also include `server` instead of NULL afaik

**Comment by ailin-nemui** [comment]
really, only use "server" if the flag is given. Otherwise the printtext would curiously end up in the server window of the active item instead of the status window

**Comment by ailin-nemui** [comment]
@vague666 please rebase this and then we can merge it

**Pull request merged**